frontend/src/views/auth/Register.vue

```diff
@@ -0,0 +1,199 @@
+      <input
+        id="inputEmail"
+        v-model="email"
+        type="text"
+        name="username"
+        class="form-control"
+        placeholder="eMail"
+        required
+        autofocus>
+      <label for="inputEmail">eMail</label>
+    </div>
+
+    <div class="form-label-group">
+      <input
+        id="inputPassword1"
+        v-model="pw1"
+        type="password"
+        name="password"
+        class="form-control"
+        placeholder="Password"
+        required>
+      <label for="inputPassword1">Password</label>
+    </div>
+
+    <div class="form-label-group">
+      <input
+        id="inputPassword2"
+        v-model="pw2"
+        type="password"
+        name="password"
+        class="form-control"
+        placeholder="Password"
+        required>
+      <label for="inputPassword2">Password</label>
+    </div>
+
+    <div v-if="pwFeedback === 'ok'">
+      pw ok
+    </div>
+    <div v-if="pwFeedback === 'nok'">
+      pw error
+    </div>
+
+    <div class="form-label-group">
+      <button
+        class="btn btn-lg btn-primary btn-block"
+        type="submit"
+        :disabled="!formComplete">
+        Register
+      </button>
+    </div>
+
+    <hr style="margin-top: 40px">
+
+    <div class="form-label-group">
+      <router-link
+        :to="{ name: 'login' }"
+        class="btn btn-md btn-link btn-block">
+        Login
+      </router-link>
+    </div>
+  </form>
+</template>
+
+<script>
+export default {
+  name: 'Register',
+  data () {
+    return {
+      username: '',
+      email: '',
+      pw1: '',
+      pw2: ''
+    }
+  },
+  computed: {
+    pwFeedback () {
+      if (this.pw1 !== '' && this.pw2 !== '') {
+        return (this.pw1 === this.pw2) ? 'ok' : 'nok'
+      }
+      return ''
+    },
+    formComplete () {
+      return (this.username !== '') && (this.email !== '') &&
+        (this.pw1 !== '') && (this.pw2 !== '') &&
+        (this.pw1 === this.pw2)
+    },
+    formData () {
+      return {
+        username: this.username,
+        email: this.email,
+        pw: this.pw1
+      }
+    }
+  },
+  methods: {
+    async register () {
+      let url = process.env.VUE_APP_ROOT_API + '/register'
+      let res = (await this.axios.post(url, this.formData))
+
+      if (res.status === 200) {
+        console.log(res.data)
+        this.$router.push({ name: 'register-done' })
+      } else {
+        console.warn(res.data)
+      }
+    }
+  }
+}
+</script>
+
+<style>
+
+    :root {
+        --input-padding-x: .75rem;
+        --input-padding-y: .4rem;
+    }
+
+    #app {
+        width: 100%;
+        height: 100%;
+    }
+    .form-register {
+        width: 100%;
+        max-width: 420px;
+        padding: 15px;
+        margin: 0 auto;
+    }
+    .form-label-group {
+        position: relative;
+        margin-bottom: 1rem;
+    }
+    .form-label-group > input,
+    .form-label-group > label {
+        padding: var(--input-padding-y) var(--input-padding-x);
```

Comment:
Ich habe es noch nicht lokal laufen gelassen, aber warum all diese (im anderen File duplizierten) Form Style Anpassungen? Gibt es keine Klassen in Bootstrap mit denen man das erreichen könnte?

Reply:
Nur ein Entwurf, Code ist geklaut ;)